include/event.php

```diff
@@ -516,9 +535,9 @@ function event_by_id($owner_uid = 0, $event_params, $sql_extra = '') {
 		intval($event_params["event_id"])
 	);
 
-	if (dbm::is_result($r))
-		return $r;
-
+	if (dbm::is_result($r)) {
+		return event_remove_duplicates($r);
```

Comment:
Well one question: wouldn't it be better to, remove the duplicated entries from the db and avoid the creation of duplicates than to remove the duplicates every time from the array?